code-react/src/views/dashboard/app/user-profile-edit.js

```diff
@@ -44,206 +114,299 @@ const UserProfileEdit =() =>{
-                                                  Zip Code: 85001">
-                                                  </Form.Control>
-                                              </Form.Group>
-                                          </Row>
-                                          <Button type="submit" className="btn btn-primary me-2">Submit</Button>
-                                          <Button type="reset" className="btn bg-soft-danger">Cancel</Button>
-                                      </Form>
-                                  </Card.Body>
-                              </Card>
-                          </Tab.Pane>
-                          <Tab.Pane eventKey="second" className="fade show">
+                          <UserInfo userInput = {userInput} userInfo={userInfo} onUserInput={setUserInput} updateUser={updateUser} refetch = {refetch}/>
+
+                          <ChangePassword />
+
+                          {familyMembers.Members.map((member) => <FamilyMember key={member.id} member={member} />)}
+
+
+                          <ManageContact />   
+                                               
+                      </Tab.Content>
+                  {/* </div> */}
+              </Col>
+          </Row>
+          </Tab.Container>
+        </Container>
+      </>
+  )
+
+}
+
+
+function UserInfo({ userInput, onUserInput, userInfo, updateUser, refetch}){
+    const [isEditMode, setIsEditMode] = useState(false);
+    const toggleEditMode = () => {
+        setIsEditMode(!isEditMode);
+      };
+
+    const handleCancel = () => {
+        // Reset userInput to the original userInfo values
+        onUserInput({ ...userInfo });
+        setIsEditMode(false);
+      };
+    
+    // function getChangedData(original, updated) {
+    //     const changes = {};
+    //     Object.keys(updated).forEach(key => {
+    //         // Check if the property exists in the original data and if it has changed
+    //         if (original[key] !== updated[key]) {
+    //             changes[key] = updated[key];
+    //         }
+    //     });
+    //     return changes;
+    // }
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        if (userInfo.role === 'parent' && userInput.role !== 'parent') {
+            toast.error('Parent role cannot be changed.');
+            return;
+        }
+
+        // const changed = getChangedData(userInfo, userInput);
+        // Construct the payload with currentUserName and the updated fields
+        const payload = {
+            ...userInput, // Spread the updated userInput fields
+            currentUserName: userInfo.userName, // Include the original userName to find the user
+        };
+        console.log("Updating user with payload:", payload);
+    
+        try {
+            await updateUser(payload).unwrap();
+            toast.success('User updated successfully!');
+            refetch();

```

Comment:
Important note here, you  dont need to call refetch because the update endpoint will returned the updated user 
make it like updatedUser = await updateUser(payload).unwrap();
then set that updatedUser 

Same when updating another family member